app/controllers/casework/documents_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Casework
+  class DocumentsController < Casework::BaseController
+    before_action :set_crime_application, :set_document
+
+    class CannotDownloadUnlessAssigned < StandardError; end
+    class CannotDownloadDocNotPartOfApp < StandardError; end
+
+    def show; end
+
+    # rubocop:disable Metrics/AbcSize
+    def download
```

Comment:
It feels a bit odd raising and rescuing in the same method. I wonder if it would be better to have a before filter checks. Something like: "require_assigned_user".

require_assigned_user method would then do check that document is from current_application and current_application is assigned to user.

Reply:
Nice I've made that change